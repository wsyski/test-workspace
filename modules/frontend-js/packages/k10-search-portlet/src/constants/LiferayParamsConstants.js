export const VIEW_MODE_FULL = 'FULL';
export const VIEW_MODE_BRIEF = 'BRIEF';

export const BASE_URL = 'baseUrl';
export const SEARCH_PAGE = 'searchPage';
export const SEARCH_QUERY_PARAMETER = 'searchQueryParameter';
export const COLLECTION = 'collection';
export const PAGE_SIZE = 'pageSize';
export const VIEW_MODE = 'viewMode';

export const PORTLET_INSTANCE_DEFAULT = {
	[BASE_URL]: 'https://bookit.axiell.com/k10/index/axiell',
	[COLLECTION]: [
		'medline',
		'NL',
		'JSTOR',
		'springer',
		'DOAJ',
		'elsevier',
		'CTG',
	],
	[PAGE_SIZE]: 8,
	[SEARCH_PAGE]: 'k10-search',
	[SEARCH_QUERY_PARAMETER]: 'p_r_p_arena_urn:arena_search_query',
	[VIEW_MODE]: VIEW_MODE_FULL,
};
