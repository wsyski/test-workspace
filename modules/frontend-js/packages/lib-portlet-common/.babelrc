{
	"presets": [
		"@babel/preset-env",
		"@babel/preset-react"
	],
	"plugins": [
		"@babel/plugin-proposal-class-properties",
		"@babel/plugin-proposal-nullish-coalescing-operator"
	],
	"ignore": [
		"jest.config.js",
		"src/setupTests.js",
		"**/__mocks__/*",
		"**/__tests__/*"
	]
}
