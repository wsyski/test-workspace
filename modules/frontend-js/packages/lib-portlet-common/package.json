{
  "name": "@arena/lib-portlet-common",
  "version": "1.0.0",
  "description": "Portlet common library",
  "license": "UNLICENSED",
  "private": true,
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/index.d.ts",
  "source": "src/index.ts",
  "engines": {
    "node": "=16",
    "yarn": "1.x"
  },
  "dependencies": {
    "@arena/portal": "*",
    "@liferay/portal-7.4": "*",
    "i18next": "^22.4.13",
    "i18next-chained-backend": "^4.2.0",
    "i18next-localstorage-backend": "^4.1.0",
    "i18next-xhr-backend": "^3.2.2",
    "react-i18next": "~10.13.2"
  },
  "devDependencies": {
    "@babel/runtime": "^7.21.0",
    "@babel/plugin-transform-runtime": "^7.21.0",
    "@babel/preset-env": "^7.21.4",
    "@babel/preset-react": "^7.18.6",
    "@rollup/plugin-babel": "^6.0.3",
    "@rollup/plugin-commonjs": "^24.0.1",
    "@rollup/plugin-terser": "^0.4.0",
    "@rollup/plugin-typescript": "^11.0.0",
    "@testing-library/jest-dom": "~5.16.4",
    "@testing-library/react": "~12.1.5",
    "@testing-library/user-event": "~12.8.3",
    "@types/enzyme": "~3.10.12",
    "@types/enzyme-adapter-react-16": "~1.0.6",
    "@types/jest": "~27.5.1",
    "@types/node": "^16.11.27",
    "@types/react": "16.9.56",
    "@types/react-dom": "16.9.14",
    "@types/react-redux": "7.1.25",
    "@types/react-router": "5.1.18",
    "@types/react-router-dom": "5.3.3",
    "@types/react-test-renderer": "~16.9.3",
    "@types/warning": "~3.0.0",
    "enzyme": "~3.11.0",
    "enzyme-adapter-react-16": "~1.15.6",
    "enzyme-to-json": "~3.6.1",
    "event-source-polyfill": "~1.0.25",
    "jest": "~27.5.1",
    "jest-environment-jsdom": "~27.5.1",
    "react-test-renderer": "16.12.0",
    "rollup": "^3.18.0",
    "ts-jest": "^27.1.5",
    "typescript": "~4.1.6"
  },
  "scripts": {
    "deploy": "true",
    "clean": "rimraf dist",
    "build": "rollup -c",
    "build:prod": "cross-env NODE_ENV=production rollup -c",
    "test": "jest --verbose",
    "tsc": "tsc -b tsconfig.json",
    "publish": "true"
  },
  "files": [
    "dist"
  ]
}
