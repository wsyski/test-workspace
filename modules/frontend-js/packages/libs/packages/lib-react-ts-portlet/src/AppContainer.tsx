/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {
	ALERT_ORIGIN_DEFAULT,
	Alert,
	AlertActions,
	AlertActionsContext,
	AlertMessages,
	AlertUtil,
	AppContainerProps,
	ErrorBoundary,
	LiferayParamsContext,
	LiferayUtil,
} from '@arena/lib-portlet-common';
import {ClayIconSpriteContext} from '@clayui/icon';
import React, {Suspense, useImperativeHandle, useRef} from 'react';

import App from './components/App';

interface AlertPanelProps {}

const AlertPanel = React.forwardRef(
	(props: AlertPanelProps, ref: React.Ref<AlertActions>) => {
		const [alerts, setAlerts] = React.useState<Alert[]>([]);

		const alertActions = AlertUtil.getAlertActions(setAlerts);

		useImperativeHandle(ref, () => alertActions);

		const onConfirm = React.useCallback(
			(alert: Alert) => {
				switch (alert.origin) {
					case ALERT_ORIGIN_DEFAULT:
						alertActions.deleteAlert(alert);
						break;
					default:
						throw new Error(
							'Unexpected alert origin: ' + alert.origin
						);
				}
			},
			[alertActions]
		);

		return <AlertMessages alerts={[...alerts]} onConfirm={onConfirm} />;
	}
);

const AppContainer: React.FC<AppContainerProps> = ({
													   liferayParams,
												   }: AppContainerProps) => {
	const alertActionsRef = useRef<AlertActions>(null);

	return (
		<div data-testid="appContainer">
			<ClayIconSpriteContext.Provider
				value={LiferayUtil.getClaySpritemap()}
			>
				<LiferayParamsContext.Provider value={liferayParams}>
					<Suspense fallback={<></>}>
						<AlertPanel ref={alertActionsRef} />

						<ErrorBoundary>
							<AlertActionsContext.Provider
								value={alertActionsRef}
							>
								<App />
							</AlertActionsContext.Provider>
						</ErrorBoundary>
					</Suspense>
				</LiferayParamsContext.Provider>
			</ClayIconSpriteContext.Provider>
		</div>
	);
};

export default AppContainer;
