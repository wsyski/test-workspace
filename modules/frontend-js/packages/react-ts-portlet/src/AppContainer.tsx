import {
	ALERT_ORIGIN_DEFAULT,
	Alert,
	AlertActions,
	AlertActionsContext,
	AlertMessages,
	AlertUtil,
	AppContainerProps,
	ErrorBoundary,
	LiferayParamsContext,
	LiferayUtil,
} from '@arena/lib-portlet-common';
import {ClayIconSpriteContext} from '@clayui/icon';
import React, {Suspense, useImperativeHandle, useRef} from 'react';

import App from './components/App';

interface AlertPanelProps {}

const AlertPanel = React.forwardRef(
	(props: AlertPanelProps, ref: React.Ref<AlertActions>) => {
		const [alerts, setAlerts] = React.useState<Alert[]>([]);

		const alertActions = AlertUtil.getAlertActions(setAlerts);

		useImperativeHandle(ref, () => alertActions);

		const onConfirm = React.useCallback(
			(alert: Alert) => {
				switch (alert.origin) {
					case ALERT_ORIGIN_DEFAULT:
						alertActions.deleteAlert(alert);
						break;
					default:
						throw new Error(
							'Unexpected alert origin: ' + alert.origin
						);
				}
			},
			[alertActions]
		);

		return <AlertMessages alerts={[...alerts]} onConfirm={onConfirm} />;
	}
);

const AppContainer: React.FC<AppContainerProps> = ({
													   liferayParams,
												   }: AppContainerProps) => {
	const alertActionsRef = useRef<AlertActions>(null);

	return (
		<div data-test-id="appContainer">
			<ClayIconSpriteContext.Provider
				value={LiferayUtil.getClaySpritemap()}
			>
				<LiferayParamsContext.Provider value={liferayParams}>
					<Suspense fallback={<React.Fragment />}>
						<AlertPanel ref={alertActionsRef} />

						<ErrorBoundary>
							<AlertActionsContext.Provider
								value={alertActionsRef}
							>
								<App />
							</AlertActionsContext.Provider>
						</ErrorBoundary>
					</Suspense>
				</LiferayParamsContext.Provider>
			</ClayIconSpriteContext.Provider>
		</div>
	);
};

export default AppContainer;
