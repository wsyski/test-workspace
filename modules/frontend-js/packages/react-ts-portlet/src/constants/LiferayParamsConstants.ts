const PORTLET_NAMESPACE_DEFAULT = '_portlet_namespace_';

export const LIFERAY_PARAMS_DEFAULT = {
	configuration: {
		portletInstance: {},
		system: {},
	},
	contextPath: '',
	portletElementId: 'js-portlet-' + PORTLET_NAMESPACE_DEFAULT,
	portletNamespace: PORTLET_NAMESPACE_DEFAULT,
};

const DRINK = 'drink';

export const PORTLET_INSTANCE_DEFAULT = {
	[DRINK]: 'orange',
};
