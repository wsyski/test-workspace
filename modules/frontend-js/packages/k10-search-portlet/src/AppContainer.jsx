import {LiferayParamsContext, LiferayUtil} from '@arena/lib-portlet-common';
import {ClayIconSpriteContext} from '@clayui/icon';
import React from 'react';
import {HashRouter as Router, Route, Switch} from 'react-router-dom';

import App from './App';
import RecordWrapper from './components/RecordWrapper';
import SearchResultWrapper from './components/SearchResultWrapper';

const AppContainer = ({liferayParams}) => {
	return (
		<div data-test-id="appContainer">
			<ClayIconSpriteContext.Provider
				value={LiferayUtil.getClaySpritemap()}
			>
				<LiferayParamsContext.Provider value={liferayParams}>
					<App>
						<Router>
							<Switch>
								<Route
									path="/record/:id"
									render={(props) => {
										return (
											<RecordWrapper
												id={props.match.params.id}
												{...props}
											/>
										);
									}}
								/>

								<Route
									component={SearchResultWrapper}
									exact
									path=""
								/>
							</Switch>
						</Router>
					</App>
				</LiferayParamsContext.Provider>
			</ClayIconSpriteContext.Provider>
		</div>
	);
};

export default AppContainer;
