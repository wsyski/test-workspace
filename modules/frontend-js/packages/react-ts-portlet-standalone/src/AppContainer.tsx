import {ClayIconSpriteContext} from '@clayui/icon';
import React from 'react';

import App from './components/App';
import {AppContainerProps} from "./lib-portlet-common";
import LiferayUtil from "./utils/LiferayUtil";

import LiferayParamsContext from './contexts/LiferayParamsContext';

const AppContainer: React.FC<AppContainerProps> = ({
                                                       liferayParams,
                                                   }: AppContainerProps) => {

    return (
        <div data-test-id="appContainer">
            <ClayIconSpriteContext.Provider value={LiferayUtil.getClaySpritemap()}>
                <LiferayParamsContext.Provider value={liferayParams}>
                    <App/>
                </LiferayParamsContext.Provider>
            </ClayIconSpriteContext.Provider>
        </div>
    );
};

export default AppContainer;
