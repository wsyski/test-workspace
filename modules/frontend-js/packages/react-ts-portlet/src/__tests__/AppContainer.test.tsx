import {LIFERAY_PARAMS_DEFAULT, LiferayUtil} from '@arena/lib-portlet-common';
import {ShallowWrapper, shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import AppContainer from '../AppContainer';
import {PORTLET_INSTANCE_DEFAULT} from '../constants/LiferayParamsConstants';

const findShallowByTestAttr = (
	component: ShallowWrapper,
	attr: string
): ShallowWrapper => {
	return component.find(`[data-test-id='${attr}']`);
}

describe('AppContainer Component', () => {
	let wrapper: ShallowWrapper;
	beforeEach(() => {
		jest.spyOn(LiferayUtil, 'getScopeGroupId').mockImplementation(
			() => 39847
		);
		jest.spyOn(LiferayUtil, 'getPortalURL').mockImplementation(
			() => 'http://localhost:6080'
		);
		jest.spyOn(LiferayUtil, 'getPathThemeImages').mockImplementation(
			() => '/theme/images'
		);
		jest.spyOn(LiferayUtil, 'getLocale').mockImplementation(() => 'en_US');
		const liferayParamsWithDefaults = LiferayUtil.setLiferayParamsDefaults(
			LIFERAY_PARAMS_DEFAULT,
			PORTLET_INSTANCE_DEFAULT
		);
		wrapper = shallow(
			<AppContainer liferayParams={liferayParamsWithDefaults} />
		);

		// console.log(wrapper.debug());
	});

	afterEach(() => {
		jest.clearAllMocks();
		wrapper.unmount();
	});

	test('AppContainer should mount without errors', () => {
		const component = findShallowByTestAttr(
			wrapper,
			'appContainer'
		);
		expect(component.length).toBe(1);
	});

	test('matches snapshot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
