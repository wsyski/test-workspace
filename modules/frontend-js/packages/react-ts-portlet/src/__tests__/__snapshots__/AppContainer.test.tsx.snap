// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer Component matches snapshot 1`] = `
<div
  data-test-id="appContainer"
>
  <ContextProvider
    value="/theme/images/lexicon/icons.svg"
  >
    <ContextProvider
      value={
        Object {
          "configuration": Object {
            "portletInstance": Object {
              "drink": "orange",
            },
            "system": Object {},
          },
          "contextPath": "",
          "portletElementId": "js-portlet-_portlet_namespace_",
          "portletNamespace": "_portlet_namespace_",
        }
      }
    >
      <Suspense
        fallback={<React.Fragment />}
      >
        <ForwardRef />
        <withI18nextTranslation(ErrorBoundaryWithT)>
          <ContextProvider
            value={
              Object {
                "current": null,
              }
            }
          >
            <App />
          </ContextProvider>
        </withI18nextTranslation(ErrorBoundaryWithT)>
      </Suspense>
    </ContextProvider>
  </ContextProvider>
</div>
`;
